### PR TITLE
add [AutoResolve] for reading from app setting

### LINKED ISSUE #200: Add support for app settings binding expressions
Can you please add support for binding expression in SQL command text

### PR DESCRIPTION
**Pull request by MaddyDev**
Adding  [AutoResolve] tag enables reading the %<appKey>% values from app config during run time. This was already added to parameters but missed from CommandText. Adding it to enable reading stored procedure values from app config.
This fixes #200.

### CODE REVIEW

**Review by Charles-Gagnon — COMMENTED** [review]
Please add a sample and test for this

**Review by lucyzhang929 — APPROVED** [review]

File: samples/Common/ProductUtilities.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public static List<Product> GetNewProducts(int num)
                 var product = new Product
                 {
                     ProductID = i,
-                    Cost = 100,
+                    Cost = i,
```
Comment (chlafreniere) [inline samples/Common/ProductUtilities.cs:22]: do we have to change this as well?

Comment (MaddyDev) [reply samples/Common/ProductUtilities.cs:22]: not really, but to query I thought it would be better to have different values.

Comment (Charles-Gagnon) [reply samples/Common/ProductUtilities.cs:22]: Let's at least make the cost `100 + i` or something to clearly differentiate it from the ID

File: samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.Sql.Samples.Common;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Samples.InputBindingSamples
+{
+
+    public static class GetProductsStoredProcedureFromAppSetting
```
Comment (Charles-Gagnon) [inline samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs:13]: Please add a test that verifies this route as well

File: samples/local.settings.json [hunk 3]
```diff
@@ -3,6 +3,8 @@
   "Values": {
     "AzureWebJobsStorage": "UseDevelopmentStorage=true",
     "FUNCTIONS_WORKER_RUNTIME": "dotnet",
-    "SqlConnectionString": ""
+    "SqlConnectionString": "",
+    "Sp_SelectCost": "SelectProductsCost",
+    "Cost100": 100
```
Comment (Charles-Gagnon) [inline samples/local.settings.json:8]: Let's try to keep the samples as generic as possible - so for this I'd suggest `ProductCost` instead

File: samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        [FunctionName("GetProductsStoredProcedureFromAppSetting")]
+        public static IActionResult Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "getproductscost100-storedprocedure-appsetting")]
```
Comment (Charles-Gagnon) [inline samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs:17]: Along with updating the app setting name I'd suggest we name the route something more generic like `getproductsbycost`.

The route really we should just be thinking of from the users point of view - what makes sense for someone calling this API. We have other ways like the function name to indicate to the developer looking at this sample what this sample is demonstrating.

File: samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.Sql.Samples.Common;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Samples.InputBindingSamples
+{
+
+    public static class GetProductsStoredProcedureFromAppSetting
+    {
+        [FunctionName("GetProductsStoredProcedureFromAppSetting")]
+        public static IActionResult Run(
```
Comment (Charles-Gagnon) [inline samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs:16]: Please add a doc comment explaining what this function is demonstrating

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.Sql.Samples.Common;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Samples.InputBindingSamples
+{
+    /// <summary>
+    /// This shows an example of a SQL Input binding that uses a store procedure from app setting
```
Comment (Charles-Gagnon) [inline samples/InputBindingSamples/GetProductsStoreProcedureFromAppSetting.cs:13]:
```
This shows an example of a SQL Input binding that uses a stored procedure from an app setting value to query for Products with a specific cost that is also defined as an app setting value.
```

File: test/Integration/SqlInputBindingIntegrationTests.cs [hunk 7]
```diff
@@ -98,6 +98,25 @@ public async void GetProductsNameEmptyTest(int n, int cost)
             Assert.Equal(expectedResponse, actualResponse, StringComparer.OrdinalIgnoreCase);
         }
 
+        [Fact]
+        public async void GetProductsByCostTest()
+        {
+            this.StartFunctionHost(nameof(GetProductsStoredProcedureFromAppSetting));
+
+            // Generate T-SQL to insert n rows of data with cost
+            Product[] products = GetProductsWithSameCost(1, 100);
```
Comment (Charles-Gagnon) [inline test/Integration/SqlInputBindingIntegrationTests.cs:107]: You should also insert some products with cost != 100 to make sure we're only grabbing the expected products

**Review by Charles-Gagnon — APPROVED** [review]
Few more cleanup things, but overall looks good!

**Pull request merged**